pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -491,6 +491,12 @@ func (s *BaseScheduler) updateTask(taskState proto.TaskState, newSubTasks []*pro
 		if err == nil || !retryable {
 			break
 		}
+		select {
+		case <-s.ctx.Done():
```

Comment:
if err := s.ctx.Err(); err != nil {
  return err
}

Reply:
also for below code

Reply:
d0a5a2c8bbbc4c80cd88906ec4d0d8f792f91fc8

Reply:
no I mean the whole select can be replaced